backbone/src/main/java/org/researchstack/backbone/model/survey/SurveyItem.java

```diff
@@ -54,4 +51,30 @@ public int compare(SurveyItemType lhs, SurveyItemType rhs) {
             return 0;
         }
     }
+
+    @Override
+    public int hashCode() {
+        if (identifier == null) {
```

Comment:
Good question... I wish I had a better answer than, "its all I needed to get unit tests and the code working"

What else do you think would be appropriate for comparison?